### PR TITLE
Header and minor ui changes

### PR DESCRIPTION
**Pull request by apollo1291**
Adds the Header component (nav bar at the top of the page)

**Note: we don't automatically generate profile pictures for new accounts, so the alt text shows in the image below. We should either generate this in the backend (https://github.com/battlecode/galaxy/pull/327 has some work on this), or display a default profile picture**. I've added this onto our roadmap

Testing:
* The hamburger menu (left side of header) should appear at small widths, and the side bar should disappear. The hamburger menu's dropdown should contain all the items from the side bar. 
  * When logged out, there should be a `Log in` option at the bottom of the dropdown.
* When logged in, the profile image should always appear in the right side of the header. Clicking the profile image triggers a dropdown with 2 options: `Your profile`, and `Sign out`
* When logged out, there should be a `Log in` and `Register` button in the right side of the header.  The `Log in` button will disappear at small window widths.

image
image
image
image

### CODE REVIEW

File: frontend2/src/components/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+
+interface headerProps {
+    loggedIn: boolean
+    logoPath: string
+}
+const Header = ({loggedIn, logoPath}: headerProps): JSX.Element => {
```
Comment (acrantel) [inline frontend2/src/components/Header.tsx:7]: for this too, let's be consistent with how we defined basecard
`const BaseCard: React.FC<BaseCardProps> = ({ GameImage, GameName, GameDescr, GameYear }) => {`

File: frontend2/src/components/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+
+interface headerProps {
```
Comment (acrantel) [inline frontend2/src/components/Header.tsx:3]:
```suggestion
interface HeaderProps {
```
for consistency

File: frontend2/src/components/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    loggedIn: boolean
```
Comment (acrantel) [inline frontend2/src/components/Header.tsx:4]: This is OK for now, we will switch to a context for user information later

**Review by acrantel — CHANGES_REQUESTED** [review]

File: frontend2/src/components/BaseCard.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+interface BaseCardProps {
+  GameImage: string;
```
Comment (acrantel) [inline frontend2/src/components/BaseCard.tsx:4]: let's change all these properties to camelCase

`GameImage` => `gameImage`

etc

**Comment by acrantel** [comment]
https://tailwindui.com/components/application-ui/application-shells/stacked

File: frontend2/src/battlecode-logo-horiz-white.png
Comment (acrantel) [inline frontend2/src/battlecode-logo-horiz-white.png:1]: this file doesn't belong in the src folder, all images should go in the `frontend2/public` folder

File: frontend2/src/components/BaseCard.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+const BaseCard: React.FC<BaseCardProps> = ({ gameImage, gameName, gameDescr, gameYear }): JSX.Element => {
+  const redirect = (): void => {
+    console.log(gameYear.toString()); // Redirect to year page
+  };
+
+  return (
+    <div className="max-w-sm rounded overflow-hidden shadow-md transition duration-300 ease-in-out transform hover:shadow-lg hover:cursor-pointer" onClick={redirect}>
+      <div className="aspect-w-4 aspect-h-3">
+        <img className="object-cover" src={require(`../${gameImage}`)} alt={gameName} />
```
Comment (acrantel) [inline frontend2/src/components/BaseCard.tsx:18]:
```suggestion
        <img className="object-cover" src={gameImage} alt={gameName} />
```
you don't need the require() here

File: frontend2/src/components/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+                stroke="currentColor"
+                aria-hidden="true"
+              >
+                <path strokeLinecap="round" strokeLinejoin="round" d="M6 18L18 6M6 6l12 12" />
+              </svg>
+            </button>
+          </div>
+          <div className="flex flex-1 items-center justify-center sm:items-stretch sm:justify-start">
+            <div className="flex flex-shrink-0 items-center">
+              <img className="h-8 w-auto" src={require(`../${logoPath}`)} alt="Battlecode" />
```
Comment (acrantel) [inline frontend2/src/components/Header.tsx:63]: similar to above, you don't need the require here.

File: frontend2/src/components/Header.tsx [hunk 7]
```diff
@@ -0,0 +1,345 @@
...
+            </div>
+            <div className="hidden sm:ml-6 sm:block">
+        {/* <div className="flex space-x-4">
```
Comment (acrantel) [inline frontend2/src/components/Header.tsx:66]: delete these big comments of code if we don't need them

File: frontend2/src/components/Header.tsx [hunk 8]
```diff
@@ -0,0 +1,345 @@
...
+            <button
+              type="button"
+              className="relative rounded-full bg-gray-800 p-1 text-gray-400 hover:text-white focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-gray-800"
+            >
+              <span className="absolute -inset-1.5"></span>
+              <span className="sr-only">View notifications</span>
+              <svg className="h-6 w-6" fill="none" viewBox="0 0 24 24" strokeWidth="1.5" stroke="currentColor" aria-hidden="true">
+                <path strokeLinecap="round" strokeLinejoin="round" d="M14.857 17.082a23.848 23.848 0 005.454-1.31A8.967 8.967 0 0118 9.75v-.7V9A6 6 0 006 9v.75a8.967 8.967 0 01-2.312 6.022c1.733.64 3.56 1.085 5.455 1.31m5.714 0a24.255 24.255 0 01-5.714 0m5.714 0a3 3 0 11-5.714 0" />
+              </svg>
+            </button>
```
Comment (acrantel) [inline frontend2/src/components/Header.tsx:120]: This notification button doesn't seem to do anything right now, let's remove it.

File: frontend2/src/components/Header.tsx [hunk 9]
```diff
@@ -0,0 +1,345 @@
...
+            aria-orientation="vertical"
+            aria-labelledby="user-menu-button"
+            tabIndex={-1}
+          >
+            <a href="#" className="block px-4 py-2 text-sm text-gray-700" role="menuitem" tabIndex={-1} id="user-menu-item-0">
+              Your Profile
+            </a>
+            <a href="#" className="block px-4 py-2 text-sm text-gray-700" role="menuitem" tabIndex={-1} id="user-menu-item-1">
+              Settings
+            </a>
```
Comment (acrantel) [inline frontend2/src/components/Header.tsx:155]: We don't have a settings page right now, let's remove this

File: frontend2/src/components/Header.tsx [hunk 10]
```diff
@@ -0,0 +1,345 @@
...
+        Scrimaging
+      </a>
+    </div>
+  </div>}
+    </nav>
+  );
+};
+
+export default Header;
+// const Header: React.FC<headerProps> = ({loggedIn, logoPath}): JSX.Element => {
```
Comment (acrantel) [inline frontend2/src/components/Header.tsx:310]: Let's delete this big comment if it's unused.

File: frontend2/src/components/Header.tsx [hunk 11]
```diff
@@ -0,0 +1,345 @@
...
+            className="relative flex rounded-full bg-gray-800 text-sm focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-gray-800"
+            id="user-menu-button"
+            aria-expanded={isUserMenuOpen}
+            aria-haspopup="true"
+          >
+            <span className="absolute -inset-1.5"></span>
+            <span className="sr-only">Open user menu</span>
+            <img
+              className="h-8 w-8 rounded-full"
+              src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
```
Comment (acrantel) [inline frontend2/src/components/Header.tsx:136]: This is OK to keep for this PR. Eventually, we will want to fetch the user avatar and put it here instead.

**Review by acrantel — CHANGES_REQUESTED** [review]
thanks for tackling this! headers are super complicated

I put this in comments but also wanted to mention it here; images should live in the `frontend2/public` directory. When you want to reference it in code, you just need to use `/<image_name_with_extension>` to display it

**Review by lowtorola — APPROVED** [review]
LGTM! 🎉

**Pull request merged**